From c5ddc72852c49787c331870c128453e0d07f22fb Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 24 Oct 2019 16:12:06 -0500
Subject: [PATCH] event: ensure alarms have an ACTION, WIP

---
 src/event.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/event.c b/src/event.c
index a807723..ba9efc5 100644
--- a/src/event.c
+++ b/src/event.c
@@ -188,6 +188,13 @@ void event_set_alarm_trigger(Event* ev, const char* trigger_string)
 		icalcomponent_add_property(valarm, icalproperty_new_trigger(trigger));
 	}
 
+  // VALARM must have exactly one ACTION (or so my server insists)
+  // DISPLAY seems like a good default, but requires a description FWIW
+  //( Not sure if allowed to set NONE?)
+  enum icalproperty_action action = ICAL_ACTION_DISPLAY;
+  icalproperty* action_prop = icalproperty_new_action(action);
+  icalcomponent_add_property(valarm, action_prop);
+
 	ev->dirty = TRUE;
 }
 
-- 
2.24.0-rc1

