From 424bde062578c9449487958f1806f7c9730604da Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Mon, 28 Oct 2019 06:58:12 -0500
Subject: [PATCH] focal: print put string

---
 src/caldav-calendar.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/caldav-calendar.c b/src/caldav-calendar.c
index 255a3d1..22b1b1c 100644
--- a/src/caldav-calendar.c
+++ b/src/caldav-calendar.c
@@ -310,6 +310,7 @@ static void do_caldav_put(CaldavCalendar* rc, CURL* curl, struct curl_slist* hea
 
 	curl_easy_setopt(curl, CURLOPT_CUSTOMREQUEST, "PUT");
 	ac->cal_postdata = event_as_ical_string(ac->new_event);
+  printf("ical_string: <<%s>>\n", event_as_ical_string(ac->new_event));
 
 	curl_easy_setopt(curl, CURLOPT_POSTFIELDS, ac->cal_postdata);
 	curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, strlen(ac->cal_postdata));
-- 
2.24.0-rc1

