From 943ab21973ec84060b77a7e5b7f5eac77e61851d Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Wed, 14 Nov 2018 17:05:24 -0600
Subject: [PATCH] tools: drop llvm-vr-dump (temporarily) to workaround build
 failure

---
 tools/CMakeLists.txt | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index bae73a4..818e75a 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -54,13 +54,6 @@ if (LLVM_DG)
 				PRIVATE ${llvm_analysis}
 				PRIVATE ${llvm_support})
 
-	add_executable(llvm-vr-dump llvm-vr-dump.cpp)
-	target_link_libraries(llvm-vr-dump
-				PRIVATE ${llvm_core}
-				PRIVATE ${llvm_irreader}
-				PRIVATE ${llvm_analysis}
-				PRIVATE ${llvm_support})
-
 	add_executable(llvm-to-source llvm-to-source.cpp)
 	target_link_libraries(llvm-to-source
 				PRIVATE ${llvm_core}
-- 
2.19.1

