From 6c5d18aa0e3affc11f8e8cf4efe39d5c577f2fd6 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sun, 11 Dec 2016 11:27:13 -0600
Subject: [PATCH 1/5] Makefile: workaround production of empty file on error

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3400b5c..2f1c087 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,8 @@ validate: ${HS}
 	strip validate
 
 MD/Binary.hs: bin/genBinary.hs
-	runghc bin/genBinary.hs > $@
+	runghc bin/genBinary.hs > $@.tmp
+	mv $@.tmp $@
 
 validate.hp: validate
 	ghc ${FLAGS} --make -O2 \
-- 
2.10.0

