From e649831d15461b4abf90df284fead0421efe4397 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Sun, 11 Dec 2016 11:42:37 -0600
Subject: [PATCH 4/5] fix test makefile

---
 test/sqlite/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/sqlite/Makefile b/test/sqlite/Makefile
index 9fded48..b3be2e4 100644
--- a/test/sqlite/Makefile
+++ b/test/sqlite/Makefile
@@ -23,7 +23,7 @@ clean:
 	rm -f *.[so] *.ast *.astpairs results.*
 
 sqlite3.o: sqlite3.c
-	clang -w -emit-llvm -c $<
+	clang -w -emit-llvm -c $< -o $@
 
 m2r.o: sqlite3.o
 	opt -mem2reg -f -o=$@ $<
-- 
2.10.0

